<?xml version="1.0"?>
<!--
Version:  2017.07.31
Authors:  Members of the Team NAIST-Panasonic at the Amazon Robotics Challenge 2017
          Gustavo A. Garcia R. garcia-g at is.naist.jp (Captain), 
          Lotfi El Hafi, Felix von Drigalski, Wataru Yamazaki, Viktor Hoerig, Arnaud Delmotte, 
          Akishige Yuguchi, Marcus Gall, Chika Shiogama, Kenta Toyoshima, Pedro Uriguen, 
          Rodrigo Elizalde, Masaki Yamamoto, Yasunao Okazaki, Kazuo Inoue, Katsuhiko Asai, 
          Ryutaro Futakuchi, Seigo Okada, Yusuke Kato, and Pin-Chu Yang
*******************
Copyright 2017 Team NAIST-Panasonic 
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 
    http://www.apache.org/licenses/LICENSE-2.0 
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*******************
-->

<launch>
  <!-- #### Parameters #### -->
   <arg name="tnp_debug" default="false" />
  <!-- ## topic ## -->
  <param name="tnp_grasp_planner/cloud_topic_sr300" value="/camera/depth_registered/points" />
  <!--<param name="tnp_grasp_planner/cloud_topic_sr300" value="/camera/depth/points" />-->
  <param name="tnp_grasp_planner/cloud_topic_phoxi" value="/phoxi_camera/cloudpoint" /> <!-- Photoneo -->
  <!-- ## pcd save directory ## -->
  <param name="tnp_grasp_planner/pcd_directory" value="/root/downloads_host/pcdfiles/" />
  <!-- ## surface normal estimation ## -->
  <param name="tnp_grasp_planner/ne_param/search_radius" type="double" value="0.015" />
  <param name="tnp_grasp_planner/ne_param/search_radius_suck" type="double" value="0.05" />
  <!-- ## difference extraction ## -->
  <param name="tnp_grasp_planner/de_param/octree_resolution" type="double" value="0.02" />
  <!-- ## passthrough filter ## -->
  <param name="tnp_grasp_planner/pf_param/thresh_min_z" type="double" value="0.5" />
  <param name="tnp_grasp_planner/pf_param/thresh_max_z" type="double" value="1.0" />
  <!-- ## simple/conditional Euclidean clustering ## -->
  <param name="tnp_grasp_planner/ec_param/cluster_tolerance" type="double" value="0.02" />
  <param name="tnp_grasp_planner/ec_param/min_cluster_size" type="int" value="500" />
  <!-- ## conditional Euclidean clustering ## -->
  <param name="tnp_grasp_planner/cec_param/squared_distance" type="double" value="10" />
  <param name="tnp_grasp_planner/cec_param/thresh_normal_angle_deg" type="int" value="2" />
  <param name="tnp_grasp_planner/cec_param/thresh_curvature" type="double" value="0.015" />
  <!-- ## Occupancy grid ## -->
  <param name="tnp_grasp_planner/occu_param/grid_size" type="double" value="0.01" /> <!-- not tested to change dynamically -->
  
  <!-- use this if your using depth_registered -->
  <!-- <param name="tnp_grasp_planner/occu_param/depth_camera_frame" type="string" value="tnp_ee_camera_frame" /> -->
  <!-- use this if your using MANUAL FIX -->
  <param name="tnp_grasp_planner/occu_param/depth_camera_frame" type="string" value="tnp_ee_depth_camera_frame" />

  <param name="tnp_grasp_planner/occu_param/occupied_voxel_threshold" type="double" value="0.6" />
  <param name="tnp_grasp_planner/start_searching_free_spot_from_this_z" type="double" value="0.01" />
  <param name="tnp_grasp_planner/bin_A_virtual_division_in_y" type="double" value="0.45" />
  <param name="tnp_grasp_planner/tnp_debug" type="bool" value="$(arg tnp_debug)" />
  <!-- #### Node #### -->
  <node name="tnp_grasp_planner" pkg="tnp_grasp_planner" type="tnp_grasp_planner" output="screen" />
</launch>
